latex/RULES

```diff
@@ -311,4 +163,83 @@
       }
     }
   }
+, "latexmk":
+  { "doc":
+    [ "A latexmk run"
+    , ""
+    , "Call latexmk passing main as the entry point. A correct staging"
+    , "of the needed files has to be given through the runfiles of \"deps\"."
+    ]
+  , "target_fields": ["deps", "srcs"]
+  , "string_fields": ["main", "stage", "opts", "output extension"]
+  , "field_doc":
+    { "srcs": ["The files needed for the latexmk run."]
+    , "stage":
+      [ "The directory the \"srcs\" logically reside in."
+      , "Entries are joined with \"/\"."
+      ]
+    , "deps":
+      ["Runfiles needed for the run of the standalone latex invocation"]
+    , "main":
+      [ "The entry point for the latex run; should be the base name"
+      , "of a .tex file in \"srcs\"; the \"stage\" is prepended automatically."
+      ]
+    , "opts":
+      ["The latexmk options to be used.", "If omitted, \"-pdf\" is assumed"]
+    , "output extension":
+      [ "The extension of the output to be produced (e.g., \".ps\", \".pdf\")."
+      , "If omitted, \".pdf\" is assumed."
+      ]
+    }
+  , "config_vars": ["env"]
+  , "config_doc":
+    { "env":
+      [ "Any override to the default environment which sets only"
+      , "PATH and SOURCE_DATE_EPOCH"
+      ]
+    }
+  , "imports": {"call latex": "call latex"}
+  , "expression":
+    { "type": "let*"
+    , "bindings":
+      [ ["srcs", {"type": "FIELD", "name": "srcs"}]
+      , ["main", {"type": "FIELD", "name": "main"}]
+      , ["deps", {"type": "FIELD", "name": "deps"}]
+      , ["stage", {"type": "FIELD", "name": "stage"}]
+      , [ "output file extension"
+        , { "type": "if"
+          , "cond": {"type": "FIELD", "name": "output extension"}
+          , "then": {"type": "FIELD", "name": "output extension"}
+          , "else": [".pdf"]
+          }
+        ]
+      , [ "env"
+        , {"type": "var", "name": "env", "default": {"type": "empty_map"}}
+        ]
+      , ["latex", "UNUSED"]
```

Comment:
For unused variables, the default `null` is good enough.

On the other hand, why not allow overwriting the name of the `latexmk` binary in the same way as the name of the `latex` binary can be ovewritten?

Reply:
thanks, good suggestion.

I removed the `runner type` variable. the dispatch is based on the value of `latex`